src/ConsumeOwnEventsService.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (C) Martin Da Costa 2023 (martindc.merg@gmail.com)
+// This file is part of VLCB-Arduino project on https://github.com/SvenRosvall/VLCB-Arduino
+// Licensed under the Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License.
+// The full licence can be found at: http://creativecommons.org/licenses/by-nc-sa/4.0/
+
+#include <Streaming.h>
+#include "ConsumeOwnEventsService.h"
+#include <vlcbdefs.hpp>
+
+namespace VLCB
+{
+
+void ConsumeOwnEventsService::setController(Controller *cntrl)
+{
+  this->controller = cntrl;
+  this->module_config = cntrl->getModuleConfig();
+}
+
+/// if buffer has one or more stored items
+
+bool ConsumeOwnEventsService::available(void)
+{
+  if (size > 0)
+  {
+    return true;
+  }
+  else
+  {
+  return false;
+  }
+
+ // return (size > 0);
+}
+
+/// store an item to the buffer - overwrite oldest item if buffer is full
+/// never called from an interrupt context so we don't need to worry about interrupts
+
+void ConsumeOwnEventsService::put(const VlcbMessage * msg)
+{
+//  VlcbMessage msg;
+  memcpy((VlcbMessage*)&buffer[head], (const VlcbMessage *)msg, sizeof(VlcbMessage));
```

Comment:
Why do you need the casts? (VlcbMessage*) and (const VLCBMessage *) These cast to the same type.

Reply:
That's what comes of copying things and then getting overwhelmed with joy when it finally works!  I should gone back to look for such things and taken them out.  Now changed